core/src/main/scala/kafka/server/DelayedFetch.scala

```diff
@@ -111,7 +117,8 @@ class DelayedFetch(delayMs: Long,
     }
 
     // Case D
-    if (accumulatedSize >= fetchMetadata.fetchMinBytes)
+    if (accumulatedSize >= fetchMetadata.fetchMinBytes
+      || (accumulatedThrottledSize >= fetchMetadata.fetchMinBytes && !quota.isQuotaExceeded()))
```

Comment:
If quota is not exceeded, should we check (accumulatedSize + accumulatedThrottledSize) >= fetchMetadata.fetchMinBytes?